tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Kafka/ProducerCache.cs

```diff
@@ -23,11 +27,43 @@ public static void AddProducer(object producer, string bootstrapServers)
 #endif
     }
 
+    public static void CreateDefaultDeliveryHandler<TProducer>(TProducer producer)
```

Comment:
We should probably make sure this isn't invoked if we're not using DSM (I've lost track of where this is called from)